inspirehep/modules/records/utils.py

```diff
@@ -44,3 +51,19 @@ def get_detailed_template_from_record(record):
     """Return the detailed template corresponding to the given record."""
     endpoint = get_endpoint_from_record(record)
     return current_app.config['RECORDS_UI_ENDPOINTS'][endpoint]['template']
+
+
+@contextmanager
+def file_resolving(fpath):
```

Comment:
I think this should be renamed to `resolve_file_path` or similar, an infinitive is a bad name for a function.

Reply:
Probably `open_url_or_path` is better, as it's actually returning a 'file like object' as the `open` builtin does.